src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -191,7 +181,7 @@ pub struct KvmVcpu {
     pub pio_bus: Option<devices::Bus>,
     pub mmio_bus: Option<devices::Bus>,
 
-    msr_list: HashSet<u32>,
+    msrs_to_save: HashSet<u32>,
```

Comment:
I don't remember MSRs being stored in the `KvmVcpu` structure. Is this possibly a duplicate of the data stored in the `Vm` struct?

Reply:
This is required to save snapshots. When saving snapshots, firecracker sends VcpuEvent to all vcpus and each vcpu retrieves and returns MSRs that are listed in this MSR list. At the moment of saving snapshots, vcpu thread has no longer access to `Vm` struct.

Reply:
This is still a bit odd, we still have duplicated data held in memory here. I see this field is older than this branch though, discussion is not relevant here.

Reply:
Yeah, actually I think it is a bit odd. I will try to remove these duplicated data in the PR of CPU config dump feature. Thanks for good suggestion!